sdk/eventgrid/Azure.Messaging.EventGrid/EventGridSourceGenerator/src/EventGridSourceGenerator.cs

```diff
@@ -13,38 +18,168 @@ namespace Azure.EventGrid.Messaging.SourceGeneration
     /// from constant values to deserialization method for each system event.
     /// </summary>
     [Generator]
-    internal class EventGridSourceGenerator : ISourceGenerator
+    internal class EventGridSourceGenerator : IIncrementalGenerator
     {
-        private SourceVisitor _visitor;
-        private bool _isSystemEventsLibrary;
         private const string Indent = "    ";
 
-        public void Execute(GeneratorExecutionContext context)
+        // the event name is either 3 or 4 parts, e.g. Microsoft.AppConfiguration.KeyValueDeleted or Microsoft.ResourceNotifications.HealthResources.AvailabilityStatusChanged
```

Comment:
@JoshLove-msft:  Is it safe to make this assumption?

Reply:
Yes, I think this was just refactored from https://github.com/Azure/azure-sdk-for-net/pull/49445/files#diff-9ed4376694179296155ff9439d5ef472733ab8f21a6ce3a27306e9613d5e4a07L32